src/lib/raft/raft.h

```diff
@@ -124,6 +124,12 @@ struct raft_msg {
 	 * Also is omitted when does not matter (when the message is for disk).
 	 */
 	const struct vclock *vclock;
+	/**
+	 * Whether the message is intended to be used locally or sent
+	 * to another replica. Used to figure out group_id of the xrow
+	 * (either GROUP_DEFAULT or GROUP_LOCAL).
+	 */
+	bool is_local;
```

Comment:
Wait wait, lib/raft shouldn't care about this. It only produces messages and checkpoints. Group ID is a thing made up for our replication, unrelated to the basic Raft. Lets move this all into box/raft. It is built on top of lib/raft and can deal with all the Tarantool replication specifics.

I know it seems convenient now, but lib/raft doesn't use this flag, only sets it. Which means we leak the logic from an upper module into the lower module, and that would hit us eventually if more things come into lib/raft like this, in terms of code complexity and dependencies.

Reply:
Thank you for noticing! Didn't think about that. Now it's even simpler, but I had to drop the assertions from the `raft_process_recovery`, since now, when raft state is actually received, the message has `state` and may have `vclock`.